### PR TITLE
Customizable food component

### PR DESCRIPTION
**Pull request by ghost**
:cl: qustinnus
code: adds customizable food component, doesnt do much yet
/:cl:

hello, this component is essentialy the behavior of /customizable on food into a component, I'll need this for ocne I start moving everything from the old path to the new one, it isn't applied to anything because thats impossible atm.

### CODE REVIEW

File: code/datums/components/edible.dm [hunk 1]
```diff
@@ -235,3 +247,23 @@ Behavior that's still missing from this component that original food items had t
 		if(sattisfaction_text)
 			L.emote("me", 1, "[sattisfaction_text]")
 		qdel(parent)
+
+///Tries to send a signal of type COMSIG_USE_FOOD_ON, prevents the item from attacking if a message was received (and returned COMPONENT_NO_ATTACK too)
+/datum/component/edible/proc/UseOnItem(datum/source, obj/O, mob/M, params)
+	if(SEND_SIGNAL(O, COMSIG_USE_FOOD_ON), M)
```
Comment (ShizCalev) [inline code/datums/components/edible.dm]: extra ,

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/datums/components/edible.dm (hunk 1 above)
Comment (spookydonut) [reply code/datums/components/edible.dm]: extra )

Comment (spookydonut) [inline code/datums/components/edible.dm]:
```suggestion
	if(SEND_SIGNAL(O, COMSIG_USE_FOOD_ON, M))
```

File: code/datums/components/customizable_food.dm [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
+
+	src.max_ingredients = max_ingredients
+	src.ingredients = ingredients
+	src.ingredients_placement = ingredients_placement
+	src.customname = customname
+	src.ourfood = ourfood
+	src.ourcolor = ourcolor
+
+	original_item.reagents.trans_to(parent,original_item.reagents.total_volume, transferred_by = creator)
```
Comment (spookydonut) [inline code/datums/components/customizable_food.dm]:
```suggestion
	original_item.reagents.trans_to(parent,original_item.reagents.total_volume, transfered_by = creator)
```

File: code/datums/components/customizable_food.dm [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+		I.contents += O
+
+	if(catalyst_item && creator)
+		I.attackby(catalyst_item, creator)
+
+///Destroy all of the ingredients inside.
+/datum/component/customizable_food/Destroy()
+	for(var/I in ingredients)
+		qdel(I)
+
```
Comment (spookydonut) [inline code/datums/components/customizable_food.dm]:
```suggestion
	return ..()

```

File: code/datums/components/customizable_food.dm [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+///Add all of the ingredient names together and show a length.
+/datum/component/customizable_food/proc/examine(datum/source, mob/user)
+	. = ..()
```
Comment (spookydonut) [inline code/datums/components/customizable_food.dm]:
```suggestion
```

**Review by spookydonut — CHANGES_REQUESTED** [review]

**Comment by Qustinnus** [comment]
Based on feedback I might have to rework this entirely, and instead make this an optional component on the edible component.

**Comment by spookydonut** [comment]
> 
> 
> Based on feedback I might have to rework this entirely, and instead make this an optional component on the edible component.

isn't that what ninja and i suggested to you like a week+ ago?

**Comment by Qustinnus** [comment]
yeah, I havnt updated it since then.

**Comment by Qustinnus** [comment]
I'm saying you told me to so other people know why it's stale as I'm still reowkring it

**Comment by ExcessiveUseOfCobblestone** [comment]
is this wip still

**Comment by Qustinnus** [comment]
yeah i wanted to look into fixing it but I'll just re-open it when Im finished because doing this properly in the way ninja wanted will take a while

**Comment by ExcessiveUseOfCobblestone** [comment]
:dab:

**Pull request closed without merging**